src/drivers/uwb/rddrone/rddrone.cpp

```diff
@@ -0,0 +1,698 @@
+	int option_index = 1;
+	const char *option_arg;
+	const char *device_name = nullptr;
+	bool error_flag = false;
+	int baudrate = 0;
+
+	while ((ch = px4_getopt(argc, argv, "d:b:", &option_index, &option_arg)) != EOF) {
+		switch (ch) {
+		case 'd':
+			device_name = option_arg;
+			break;
+
+		case 'b':
+			px4_get_parameter_value(option_arg, baudrate);
+			break;
+
+		default:
+			PX4_WARN("Unrecognized flag: %c", ch);
+			error_flag = true;
+			break;
+		}
+	}
+
+	if (!error_flag && device_name == nullptr) {
+		print_usage("Device name not provided.");
+		error_flag = true;
+	}
+
+	if (!error_flag && baudrate == 0) {
+		print_usage("Baudrate not provided.");
+		error_flag = true;
+	}
+
+	if (error_flag) {
+		PX4_WARN("Failed to start UWB driver.");
+		return nullptr;
+
+	} else {
+		PX4_INFO("Constructing RDDrone. Device: %s", device_name);
+		return new RDDrone(device_name, int_to_speed(baudrate));
+	}
+}
+
+int rddrone_main(int argc, char *argv[])
+{
+	return RDDrone::main(argc, argv);
+}
+
+int RDDrone::localization()
+{
+
+// 			WIP
+	/******************************************************
+	 ****************** 3D Localization *******************
+	 *****************************************************/
+
+	/*!@brief: This function calculates the 3D position of the initiator from the anchor distances and positions using least squared errors.
+	 *	 	   The function expects more than 4 anchors. The used equation system looks like follows:\n
+	 \verbatim
+	  		    -					-
+	  		   | M_11	M_12	M_13 |	 x	  b_1
+	  		   | M_12	M_22	M_23 | * y	= b_2
+	  		   | M_23	M_13	M_33 |	 z	  b_3
+	  		    -					-
+	 \endverbatim
+	 * @param distances_cm_in_pt: 			Pointer to array that contains the distances to the anchors in cm (including invalid results)
+	 * @param no_distances: 				Number of valid distances in distance array (it's not the size of the array)
+	 * @param anchor_pos: 	Pointer to array that contains anchor positions in cm (including positions related to invalid results)
+	 * @param no_anc_positions: 			Number of valid anchor positions in the position array (it's not the size of the array)
+	 * @param position_result_pt: 			Pointer to position_t variable that holds the result of this calculation
+	 * @return: The function returns a status code. */
+
+	/* 		Algorithm used:
+	 *		Linear Least Sqaures to solve Multilateration
+	 * 		with a Special case if there are only 3 Anchors.
+	 */
+
+		/* Matrix components (3*3 Matrix resulting from least square error method) [cm^2] */
+	int64_t M_11 = 0;
+	int64_t M_12 = 0;																						// = M_21
+	int64_t M_13 = 0;																						// = M_31
+	int64_t M_22 = 0;
+	int64_t M_23 = 0;																						// = M_23
+	int64_t M_33 = 0;
+
+	/* Vector components (3*1 Vector resulting from least square error method) [cm^3] */
+	int64_t b_1 = 0;
+	int64_t b_2 = 0;
+	int64_t b_3 = 0;
+
+	/* Miscellaneous variables */
+	int64_t temp;
+	int64_t temp2;
+	int64_t nominator;
+	int64_t denominator;
```

Comment:
I have no expertise to review the math here but I have to comment on the old C style variable declaration without initialization.

I always vote to keep the scope of variables as small as possible, use `const` where possible and always initialize them on first use.